src/PrestaShopBundle/Install/Install.php

```diff
@@ -892,208 +893,92 @@ public function configureShop(array $data = [])
-                }
+        foreach (scandir(_PS_MODULE_DIR_, SCANDIR_SORT_NONE) as $module) {
+            if ($module[0] != '.' && is_dir(_PS_MODULE_DIR_ . $module) && file_exists(_PS_MODULE_DIR_ . $module . '/' . $module . '.php')) {
+                $modules[] = $module;
             }
         }
 
-        return count($modules) ? $this->installModules($modules) : true;
+        return $modules;
     }
 
     /**
      * PROCESS : installModules
      * Download module from addons and Install all modules in ~/modules/ directory.
      *
-     * @param array|string|null $module Module to install. If not provided, installs all modules.
-     *
      * @return bool
      */
-    public function installModules($module = null)
+    public function installModules(): bool
     {
-        if ($module && !is_array($module)) {
-            $module = [$module];
+        $modules = $this->getModulesOnDisk();
+
+        Module::updateTranslationsAfterInstall(false);
+
+        $result = $this->executeAction($modules, 'install', 'Cannot install module "%module%"');
+        if ($result === false) {
+            return false;
         }
 
-        $modules = $module ? $module : $this->getModulesList();
+        Module::updateTranslationsAfterInstall(true);
+        EntityLanguage::updateModulesTranslations($modules);
 
-        Module::updateTranslationsAfterInstall(false);
+        return true;
+    }
 
+    public function postInstall(): bool
+    {
+        return $this->executeAction(
+            $this->getModulesOnDisk(),
+            'postInstall',
+            'Cannot execute post install on module "%module%"'
+        );
+    }
+
+    protected function executeAction(array $modules, string $action, string $errorMessage): bool
+    {
         $moduleManagerBuilder = ModuleManagerBuilder::getInstance();
         $moduleManager = $moduleManagerBuilder->build();
 
         $errors = [];
         foreach ($modules as $module_name) {
-            if (!file_exists(_PS_MODULE_DIR_ . $module_name . '/' . $module_name . '.php')) {
-                continue;
-            }
-
             $moduleException = null;
 
             try {
-                $moduleInstalled = $moduleManager->install($module_name);
-            } catch (\PrestaShopException $e) {
-                $moduleInstalled = false;
+                $moduleActionIsExecuted = $moduleManager->{$action}($module_name);
+            } catch (PrestaShopException $e) {
+                $moduleActionIsExecuted = false;
                 $moduleException = $e->getMessage();
             }
 
-            if (!$moduleInstalled) {
-                $module_errors = [$this->translator->trans('Cannot install module "%module%"', ['%module%' => $module_name], 'Install')];
+            if (!$moduleActionIsExecuted) {
+                $moduleErrors = [
+                    $this->translator->trans(
+                        $errorMessage,
+                        ['%module%' => $module_name],
```

Comment:
Just have a question regarding this string, did you remove the rest of the sentence on purpose? :)

Reply:
I -f you use a variable like this we won't be able to scan the wordings (especially the new one for post install)